src/components/ec-smart-table/ec-smart-table.spec.js

```diff
@@ -9,99 +10,188 @@ describe('EcSmartTable', () => {
     { name: 'test3', title: 'Test 3', sortable: true },
   ];
 
-  const data = [[1, 2, 3]];
+  const data = {
+    count: 3,
+    total: 3,
+    items: [[1, 2, 3]],
+  };
+
+  const emptyData = {
+    count: 0,
+    total: 0,
+    items: [],
+  };
+
+  const dataSource = {
+    fetch: jest.fn(),
+  };
 
   function mountEcSmartTable(props, mountOpts) {
     return mount(EcSmartTable, {
-      propsData: { ...props },
+      propsData: { dataSource, ...props },
+      sync: false,
       ...mountOpts,
     });
   }
 
-  function sortColumnByIndex(wrapper, index) {
-    wrapper.findByDataTest(`ec-table-head__cell--${index}`).findByDataTest('ec-table-sort__icon').trigger('click');
+  async function mountEcSmartTableWithResolvedData(resolvedData, props, mountOpts) {
+    const resolvedDataSource = {
+      fetch: jest.fn().mockResolvedValue(resolvedData),
+    };
+    const wrapper = mountEcSmartTable({ ...props, dataSource: resolvedDataSource }, mountOpts);
+    await flushPromises();
+    wrapper.vm.$forceUpdate();
```

Comment:
Out of curiosity. Why do we need to execute `$forceUpdate`? Waiting for resolution of promises was not enough in this case then?

Reply:
I couldn't figure out 100% why is this happening, but it has something to do with combination of functional and non-functional components when rendering EcSmartTable and turning the sync mode off when mounting components:
```
  withEcSmartTableContainer(
    withAbortableFetch(
      withSorting( // functional
        withEcSmartTableRenderer( // functional, including withLoading
          EcTable,
```
Without this, the table ends up always in initial loading state regardless of how many flushes/nextTicks I do.